src/functions.jl

```diff
@@ -173,6 +173,22 @@ for (T, S) in [ (:KeyedArray, :KeyedArray),
     end
 
 end
+# single argument
+Base.vcat(A::KeyedArray) = A
+function Base.hcat(A::KeyedArray)
+    data = hcat(keyless(A))
+    new_1 = keys_or_axes(A,1)
+    new_2 = ndims(A) == 1 ? axes(data,2) : keys_or_axes(A,2)
```

Comment:
I think this can be unconditional:
```suggestion
    new_2 = keys_or_axes(A,2)
```
e.g.
```
julia> A = wrapdims(rand(3), 'a':'c')
1-dimensional KeyedArray(...) with keys:
↓   3-element StepRange{Char,...}
And data, 3-element Vector{Float64}:
 ('a')  0.06664653527765108
 ('b')  0.9345180504560843
 ('c')  0.4930806744162123

julia> AxisKeys.keys_or_axes(A, 2)
Base.OneTo(1)
```

Reply:
Thanks! I've simplified a bit further. I don't have merge rights here but it's good to go from my side.